### ISSUE TITLE
How to parceler android SDK parent class

### ISSUE THREAD

**Issue by davidbilly** [johncarl81/parceler#256]
Hi @johncarl81 , Thanks for the awesome lib.

I faced some issue.

Im extend my custom with android SDK classes (non-parceable).

eg: 
reference : https://github.com/android/platform_frameworks_base/blob/master/core/java/android/app/DownloadManager.java

```java
public class Request {
    private String mTitle = "";

    public void setTitle(String title) {
        mTitle = title;
    }
} 

@Parcel
public class CustomRequest extends Request {
    private String mUrl;
    private String mFileName;

    public CustomRequest() { }
      
    public void updateTitle(String title) {
        setTitle(title);
    }
}

CustomRequest req = new CustomRequest();
req.updateTitle("Nice Title!");
```
when i passing to fragment class from activity class with bundle. 

when i received, the `CustomRequest` : `mTitle` is `""`;

Is that possible to anyway to parceable parent class at child class.

Thanks.

**Comment by johncarl81** [comment]
This should work out of the box as you have it defined.  Here's a test that passes to demonstrate:

```java
@RunWith(RobolectricTestRunner.class)
@Config(manifest=Config.NONE)
public class CustomRequestTest {

    public static class Request {
        String mTitle = "";

        public void setTitle(String title) {
            mTitle = title;
        }
    }

    @Parcel
    public static class CustomRequest extends Request {
        String mUrl;
        String mFileName;

        public CustomRequest() { }

        public void updateTitle(String title) {
            setTitle(title);
        }
    }

    @Test
    public void testRoundTrip() {
        CustomRequest request = new CustomRequest();
        request.updateTitle("Nice Title!");

        CustomRequest outputRequest = Parcels.unwrap(ParcelsTestUtil.wrap(request));

        assertEquals("Nice Title!", request.mTitle);  // Passes.
        assertEquals("Nice Title!", outputRequest.mTitle);  // Passes.

    }
}
```

Is there something else going on in your fragment, perhaps a new `CustomRequest` is being instantiated?

**Comment by johncarl81** [comment]
By the way, I'd advise you to remove `private` from your fields.

**Comment by davidbilly** [comment]
@johncarl81 , hi thanks for the test, let me try again. By the way, i cant change AOSP private field :(

actually i passing an arraylist<CustomRequest> to fragment and loop it. just when i read it, the child value is parceable(CustomRequest) just not for parent (Request) class. 

The parent class value was gone :(

**Comment by johncarl81** [comment]
I guess I don't understand your problem well enough.  Could you put together a quick example project to demonstrate the issue?

**Comment by davidbilly** [comment]
@johncarl81 sure, let me create a small project :)

**Comment by davidbilly** [comment]
@johncarl81 here's the test project. Please take a look :) 

I deleted project build resource for reduce file size, you have to re-sync project.

i test with 2 cases : 
1) MainActivity > DownloadFragment = working!
2) MainActivity > DownloadActivity > DownloadFragment = NOT working! :( 

Please assist and thank you very much.
 
[parceler.zip](https://github.com/johncarl81/parceler/files/680933/parceler.zip)

**Comment by johncarl81** [comment]
Looks like the `DownloadManager.Request` object is a stub object, or at least doesn't contain any fields, so there is nothing for Parceler to serialize.  I see you're setting the http headers that are being lost between the `MainActivity` and the `DownloadActivity`, I'd suggest writing a simple class to hold all the data you want to pass to your `DownloadActivity` then perform the `DownloadManager.Requst` creation after the fact.
